tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -5466,6 +5472,35 @@ SpirvInstruction *SpirvEmitter::createImageSample(
   return retVal;
 }
 
+void SpirvEmitter::handleOptionalTextureSampleArgs(
+    const CXXMemberCallExpr *expr, uint32_t index,
+    SpirvInstruction **constOffset, SpirvInstruction **varOffset,
+    SpirvInstruction **clamp, SpirvInstruction **status) {
+  uint32_t numArgs = expr->getNumArgs();
+
+  bool hasOffsetArg = index < numArgs &&
+                      (expr->getArg(index)->getType()->isSignedIntegerType() ||
+                       hlsl::IsHLSLVecType(expr->getArg(index)->getType()));
+  if (hasOffsetArg) {
+    handleOffsetInMethodCall(expr, index, constOffset, varOffset);
+    index++;
+  }
+
+  bool hasClampArg =
+      index < numArgs && expr->getArg(index)->getType()->isFloatingType();
+  if (hasClampArg) {
+    *clamp = doExpr(expr->getArg(index));
+    index++;
+  }
+
+  const bool hasStatusArg =
+      index < numArgs &&
+      expr->getArg(index)->getType()->isUnsignedIntegerType();
+  if (hasStatusArg) {
+    *status = doExpr(expr->getArg(index));
+  }
+}
```

Comment:
I want to clarify something. Can a call get the status, but not the clamp argument? If not, then you do not have to check the types. You could do this:

```suggestion 
  if (index >= numArgs) 
    return;
    
  *clamp = doExpr(expr->getArg(index));
  index++;
  
  if (index >= numArgs) 
    return;

  *status = doExpr(expr->getArg(index));
  }
}
```

Reply:
After rereading the proposal I believe you are correct:

> Note that these both have an additional overload that includes all the default parameters explicitly specified and a final output status parameter

I've updated to follow your suggestion.